### ISSUE TITLE
PHP Version Support

### ISSUE THREAD

**Issue by IlanAdivi1** [bdaloukas/moodle-mod_game#66]
Hello,

Hello, we are using a game plugin, and we saw that the plugin is not supported in version 8.x.

Is support for the PHP 8.X version expected?

Also, the plugin probably doesn't work well in PHP 7.4.9 version - sudoku and millionaire games.

![sudoku](https://github.com/bdaloukas/moodle-mod_game/assets/156344815/e9bf2f3f-55c7-4b6d-9526-5a902a5072cc)

Thanks,
Ilan

**Comment by bdaloukas** [comment]
It works with  PHP 7.4.9 and PHP 8.X 
Have you seen any problem to fix it?

**Comment by IlanAdivi1** [comment]
Yes, there is a problem with Sudoku and Millionaire, they don't work.

1. **sudoku** (with glossary)
https://hityashvuti.lmstest.education.gov.il/mod/game/view.php?id=1892
**glossary**
https://hityashvuti.lmstest.education.gov.il/mod/glossary/view.php?id=1891&mode=letter&hook=ALL


2. **Millionaire** (with quiz)
https://hityashvuti.lmstest.education.gov.il/mod/game/view.php?id=1895

**quiz**
https://hityashvuti.lmstest.education.gov.il/mod/quiz/edit.php?cmid=1894

**Comment by bdaloukas** [comment]
The URL https://hityashvuti.lmstest.education.gov.il/ doesn't work to do the checking.

**Comment by IlanAdivi1** [comment]
There is a problem with Sudoku and Millionaire, they don't work.

An example of sudoku that works 
![SudokuApprove](https://github.com/bdaloukas/moodle-mod_game/assets/156344815/cc3b9230-b2d9-4813-a8da-6e08e6dcd388)


Example of sudoku not working
![SudokuHit](https://github.com/bdaloukas/moodle-mod_game/assets/156344815/92cf76b7-02cb-4de5-85aa-d903dc736d37)


Millionaire
![Millionaire](https://github.com/bdaloukas/moodle-mod_game/assets/156344815/baed4aac-2491-4041-87dc-a94f54b999ed)
